tests/server.spec.js

```diff
@@ -577,6 +581,9 @@ async function expectListResponseContains(path, object) {
 }
 
 function expectStatus(response, code, type = "application/json") {
+  if (response.status === 400 && code != 400) {
```

Comment:
Nit: maybe rename the `code` arg to `expectedCode` as the function body is confusing to read